cmake/modules/versions.cmake

```
@@ -0,0 +1,72 @@
+#
+# Copyright (C) 2022 The Falco Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include(GetGitRevisionDescription)
+
+function(_get_git_version _var)
+    git_describe(tag "--tags" "--exact-match" ${ARGN})
+
+    if(tag)
+        # A tag has been found: use it as the libs version
+        set(${_var}
+            "${tag}"
+            PARENT_SCOPE)
+        return()
+    endif()
+
+    # Obtain the closest tag
+    git_describe(dev_version "--always" "--tags" "--abbrev=7" ${ARGN})
+
+    if(dev_version MATCHES "NOTFOUND$")
+        # Fallback version
+        set(dev_version "0.0.0")
+    else()
+        # Extract the git version part and make it SemVer friendly (ie. "-1-g02682d7" to "-1+02682d7" )
+        string(REGEX MATCH "(-[1-9][0-9]*-g[0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f])$"
```

I'm not 100% sure this will work in cmake, but we could simplify the regex a bit by using a quantifier
```suggestion
        string(REGEX MATCH "(-[1-9][0-9]*-g[0-9a-f]{7})$"
```

---

Hey @Molter73 

I haven't found a refernece to `{ }` in the cmake docs :point_down: 
https://cmake.org/cmake/help/latest/command/string.html#regex-specification

¯\_(ツ)_/¯

So, I'd not risk :smile_cat:

---

Yeah, agreed, It's just a shame regex often have this small convenience things that are often not implemented.